File: laravel/app/Models/Topic.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\SoftDeletes;

class Topic extends Model
{
    use HasFactory, SoftDeletes;

    protected $fillable = ['group_id', 'user_id', 'title', 'slug', 'body', 'is_pinned', 'is_locked', 'views'];

    protected $casts = ['is_pinned' => 'boolean', 'is_locked' => 'boolean'];

    public function group(): BelongsTo  { return $this->belongsTo(Group::class); }
    public function author(): BelongsTo { return $this->belongsTo(User::class, 'user_id'); }
    public function posts(): HasMany    { return $this->hasMany(Post::class); }
}

File: java-gui/src/main/java/com/smartforum/model/Topic.java
package com.smartforum.model;

public class Topic {
    public final int     id;
    public final int     groupId;
    public final int     userId;
    public final String  title;
    public final String  slug;
    public final String  body;
    public final String  authorName;
    public final boolean pinned;
    public final boolean locked;
    public final int     views;
    public final int     postsCount;

    public Topic(int id, int groupId, int userId, String title, String slug, String body,
                 String authorName, boolean pinned, boolean locked, int views, int postsCount) {
        this.id         = id;
        this.groupId    = groupId;
        this.userId     = userId;
        this.title      = title;
        this.slug       = slug;
        this.body       = body;
        this.authorName = authorName;
        this.pinned     = pinned;
        this.locked     = locked;
        this.views      = views;
        this.postsCount = postsCount;
    }

    @Override public String toString() {
        return (pinned ? "📌 " : "") + title;
    }
}
